master/internal/api_experiment.go

```diff
@@ -1955,6 +2438,14 @@ func (a *apiServer) SearchExperiments(
 		return nil, err
 	}
 
+	if req.Filter != nil {
+		filterExpr, err := parseFilter(*req.Filter)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to parse filter string: %s", err)
+		}
+		experimentQuery.Where(*filterExpr)
```

Comment:
there seems to be a risk of an SQL injection here. 
e.g. if you pass a filter `experiment.description:"' AND (SELECT password_hash FROM users WHERE id = 1)` parsed filter will be `e.config->>'description' = '' AND (SELECT password_hash FROM users WHERE id = 1)` and it'd run the injected subquery here.